### ISSUE TITLE
"Constraints related to access and use" error

### ISSUE THREAD

**Issue by sylar11** [metadata101/iso19115-3.2018#3]
Hi,
I was trying to get a fully validated ISO 19115 metadata but I was always facing this error 
**_Constraints related to access and use_**
in all the recent Geonetwork version.

I decided to dig a little bit in XSL file and I probably found the error. 
I would like to ask you to check if my suggestion is correct or if I'm misunderstanding something. 
Looking at the file 
_/core-geonetwork/web/src/main/webapp/WEB-INF/data/config/schema_plugins/iso19115-3.2018/schematron/schematron-rules-inspire.xsl_
at line 2184 is defined the variable **useLimitation_count**. 

This variable, probably, is not well defined because in my metadata I have two nodes **useLimitation_count** but if I check the value of the count function it always returns 0. 

On the contrary, if I define the variable  as follow:

`<xsl:variable name="VeB_useLimitation_count" select="count(//mdb:identificationInfo/mri:MD_DataIdentification/mri:resourceConstraints/*/mco:useLimitation/gco:CharacterString/text())"/>`
(variable name is a temporary one)

then the error disappears, the count function returns a value different from zero and the metadata ISO19115 is fully validated.

To summarize I will suggest changing this line:

l. 2184 `<xsl:variable name="useLimitation_count" select="count(mco:useLimitation/*/text())"/>`

with this one

`<xsl:variable name="useLimitation_count" select="count(//mdb:identificationInfo/mri:MD_DataIdentification/mri:resourceConstraints/mco:MD_LegalConstraints/mco:useLimitation/gco:CharacterString/text())"/>`

If you prefer to not change the variable definition, I guess that at line 2172 and following there are errors in the definition of the Xpath. 

In my opinion, the trick could be there too.

Could you please check this my suggestion?? 

Thanks in advance.

Bartolomeo

**Comment by fxprunayre** [comment]
Hi, first when you do schematron, you have to deal with *.sch files which are compiled into XSL when the application starts or when using GET /api/0.1/standards/reload in the API.

So your issue is around here https://github.com/metadata101/iso19115-3.2018/blob/3.7.x/src/main/plugin/iso19115-3.2018/schematron/schematron-rules-inspire.sch#L620.

Then indeed, if you record has 2 resource constraints, then the rule is checked on each as defined by the context.

Maybe we should change it and make it similar to the initial one in ISO19139
https://github.com/geonetwork/core-geonetwork/blob/master/schemas/iso19139/src/main/plugin/iso19139/schematron/schematron-rules-inspire.sch#L647-L674

Here the context is Identification and we check that there is at least one non empty useLimitation.

I let you check what is best and make a pull request. Thanks

**Comment by sylar11** [comment]
Thanks a lot for your answer. 
I will check your suggestion and I will let you know!!

Cheers

**Comment by sylar11** [comment]
Thanks a lot for your answer. 

Actually, my records have two "_mri:resourceConstraints_" with at least one _mco:useLimitation_ tag inside, but it is still giving me the same error.

I will check better your suggestion and I will let you know!!

Cheers
